e2etest/l2tests/interface_selector.go

```diff
@@ -107,36 +107,26 @@ var _ = ginkgo.Describe("L2-interface selector", func() {
 				Expect(err).NotTo(HaveOccurred())
 			}()
 
-			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
-			Expect(err).NotTo(HaveOccurred())
-
-			var l2Status *metallbv1beta1.ServiceL2Status
+			var l2Statuses []metallbv1beta1.ServiceL2Status
 			Eventually(func() error {
-				var ss []*metallbv1beta1.ServiceL2Status
-				if ss, err = status.GetSvcPossibleL2Status(ConfigUpdater.Client(), svc, allNodes); err == nil {
-					if len(ss) != 1 {
-						return fmt.Errorf("")
-					}
-					l2Status = ss[0]
+				var ss []metallbv1beta1.ServiceL2Status
+				if ss, err = status.GetSvcL2Status(ConfigUpdater.Client(), svc); err == nil {
+					l2Statuses = ss
 				}
 				return err
 			}, 2*time.Minute, 1*time.Second).ShouldNot(HaveOccurred())
+			Expect(l2Statuses).To(HaveLen(1))
```

Comment:
why this change? I think keeping it in the eventually is clearer (returning an error with what we got if it's not 1)